src/cdk/explorer/detectCdkProjects.ts

```diff
@@ -0,0 +1,98 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { access, stat } from '../../shared/filesystem'
+import { readDirAsString } from '../../shared/filesystemUtilities'
+
+export interface CdkProject {
+    workspaceFolder: vscode.WorkspaceFolder
+    cdkJsonPath: string
+    treePath: string
+}
+
+export async function detectCdkProjects(
+    workspaceFolders: vscode.WorkspaceFolder[] | undefined = vscode.workspace.workspaceFolders
+): Promise<CdkProject[]> {
+    if (!workspaceFolders) {
+        return []
+    }
+
+    return (await Promise.all(workspaceFolders.map(detectCdkProjectsFromWorkspaceFolder))).reduce(
+        (accumulator: CdkProject[], current: CdkProject[]) => {
+            accumulator.push(...current)
+
+            return accumulator
+        },
+        []
+    )
+}
+
+async function detectCdkProjectsFromWorkspaceFolder(workspaceFolder: vscode.WorkspaceFolder): Promise<CdkProject[]> {
+    const result = []
+
+    for await (const treeJson of detectLocalCdkProjects({ workspaceUris: [workspaceFolder.uri] })) {
+        const cdkJsonPath = path.join(treeJson.fsPath, '../', 'cdk.json')
```

Comment:
* use path.dirname anywhere you're trying to get a file's parent dir, and to go up a folder
* path.join handles slashes, it shouldn't be necessary to have one inside one of the args

Reply:
done